modules/distributor/receiver/shim.go

```diff
@@ -279,6 +291,7 @@ func (r *receiversShim) ConsumeTraces(ctx context.Context, td ptrace.Traces) err
 // ReportFatalError implements component.Host
 func (r *receiversShim) ReportFatalError(err error) {
 	_ = level.Error(log.Logger).Log("msg", "fatal error reported", "err", err)
+	r.StopAsync()
```

Comment:
I think we need to `AwaitTerminated()` here to gracefully shutdown the rest of the receivers of the shim.